src/helpers/registrars.ts

```diff
@@ -117,20 +116,22 @@ function buildContainerHandler(config: ContainerDirectiveConfig): ContainerDirec
         token.map = [params.startLine, params.endLine];
         token.markup = ':::' + config.name;
         if (container.attrs) {
-            const attrs: DirectiveAttrs = isFunction(container.attrs)
-                ? container.attrs(params)
-                : container.attrs;
+            const attrs = getValue(container.attrs, params);
             token.attrs = Object.entries(attrs);
         }
+        if (container.meta) {
+            token.meta = getValue(container.meta, params);
+        }
 
         if (inlineContent) {
             token = state.push(inlineContent.token + '_open', inlineContent.tag, 1);
             if (inlineContent.attrs) {
-                const attrs: DirectiveAttrs = isFunction(inlineContent.attrs)
-                    ? inlineContent.attrs(params)
-                    : inlineContent.attrs;
+                const attrs = getValue(inlineContent.attrs, params);
```

Comment:
nit:
It seems the code is repeated many times
This is a uniform process that can be combined into a separate function

Something like:
```ts
applyTokenField(token, 'attrs', inlineContent.attrs, params);
applyTokenField(token, 'meta', inlineContent.meta, params);

function applyTokenField<P, T>(
  token: MarkdownIt.Token,
  field: 'attrs' | 'meta',
  value: T | ((params: P) => T),
  params: P
): void {
  if (value) {
    token[field] = getValue(value, params);
    if (field === 'attrs') {
      token.attrs = Object.entries(token.attrs);
    }
  }
}
```

Reply:
👍